### PR TITLE
Add into_stream method to Page

### LINKED ISSUE #214: `impl futures::Stream for Page`
Currently paging can be a bit awkward, it would be nice if we could implement a stream API for pages.

### PR DESCRIPTION
**Pull request by mbirtwell**
More or less the easy version of what we discussed in #214.

Except that I've hidden this behind the stream feature. It's roughly orthogonal to the existing code that's behind the stream feature. But it requires exactly the same additional dependencies as that does. 

Closes #214

### CODE REVIEW

File: src/page.rs [hunk 1]
```diff
@@ -29,6 +37,13 @@ pub struct Page<T> {
     pub last: Option<Url>,
 }
 
+#[cfg(feature = "stream")]
```
Comment (XAMPPRocky) [inline src/page.rs:40]: This doesn’t need a cfg because it doesn’t use any stream specific stuff

**Review by XAMPPRocky — COMMENTED** [review]
Mostly LGTM, just some small tweaks for clarity.

File: src/page.rs (hunk 1 above)
Comment (mbirtwell) [reply src/page.rs:40]: But if we don't put the cfg on it we'll get a dead code warning, which I wanted to avoid.

Comment (XAMPPRocky) [reply src/page.rs:42]: Ah yeah, then it's fine.

**Comment by XAMPPRocky** [comment]
Thank you for your PR, and congrats on your first contribution! 🎉

**Pull request merged**